File: android/src/main/kotlin/com/timus/call_notification/models/ReceivedNotificationData.kt
package com.timus.call_notification.models

class ReceivedNotificationData(arguments: Map<String, Any>) {
    val buttonInputType : String = arguments["buttonInputType"] as String;
    val notification: NotificationData = NotificationData(arguments["notification"] as HashMap<String,Any>);

    fun toMap(): Map<String, Any> {
        var returnedObject: HashMap<String, Any> = HashMap<String, Any> ()
        returnedObject["buttonInputType"] = this.buttonInputType;
        returnedObject["notification"] = this.notification.toMap();
        return returnedObject;
    }
}

File: ios/Classes/Models/ReceivedNotificationData.swift
class ReceivedNotificationData {
    var buttonInputType: String = ""
    var notification : CallNotificationData
    
    init(json: Dictionary<String,Any>) {
        self.buttonInputType = json["buttonInputType"] as? String ?? ""
        self.notification = CallNotificationData.init(json: json["notification"] as! Dictionary<String,Any>)
    }
    
    func toMap() -> Dictionary<String,Any>{
        return [
            "buttonInputType" : self.buttonInputType,
            "notification": self.notification.toMap()
        ]
    }
}
